proposal/string-split.md

```diff
@@ -0,0 +1,105 @@
+# Add split function to Strings Module: Draft 1
+
+*([Issue](https://github.com/sass/sass/issues/1950))*
+
+This proposal adds `string.split()` to the `sass:string` module.
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+* [Semantics](#semantics)
+  * [`split()`](#split)
+
+## Background
+
+> This section is non-normative.
+
+The `sass:string` module contains several functions for
+manipulating and finding out information about strings.
+Currently, though, there is no built-in function that splits
+one string into a list of substrings, and authors have been
+creating their own versions of functions that achieve this
+functionality.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal adds the `string.split()` function to the
+`sass:string` module. The function takes a string, splits it
+based on a provided separator, and returns a space-separated
+list of substrings.
+
+This could be used to take a string and repurpose
+parts of it for some other use. For example, fonts 
+contained in a font stack list could be split into segments and 
+then used as keys in a new map. 
+
+Examples:
+
+```scss
+$fonts: "Helvetica Neue, Helvetica, Arial";
+string.split($fonts, ', '); // "Helvetica Neue" "Helvetica" "Arial"
+```
+
+A third argument can limit the number of strings 
+returned in the list:
+
+```scss
+string.split($fonts, ', ', 2); // "Helvetica Neue" "Helvetica"
+```
+
+
+An empty `$separator` returns all Unicode code points in the original string:
+
+```scss
+$font: "Helvetica"
+string.split($font, ''); // "H" "e" "l" "v" "e" "t" "i" "c" "a"
+```
+
+
+## Semantics
+
+### `split()`
+
+```
+split($string, $separator, $limit: null)
```

Comment:
Since we are returning a list here, should we allow configuration of the list's separator and brackets? Such as:

```
split ($string, $delimiter, $limit: null, $separator: auto, $bracketed: auto)
```

Reply:
Adding configuration here isn't really worth the complexity. For other functions that return a list without having an existing one to base the separator and brackets on, we always return an unbracketed, comma-separated list.

We should specify that explicitly in this proposal to match the specs of those other functions (e.g. [meta.calc-args()](https://github.com/sass/sass/blob/main/spec/built-in-modules/meta.md#calc-args)).